src/tests/functional/shared_test_classes/src/single_op/pooling.cpp

```diff
@@ -0,0 +1,170 @@
+#include "openvino/op/parameter.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/result.hpp"
+#include "openvino/op/avg_pool.hpp"
+#include "openvino/op/max_pool.hpp"
+
+namespace ov {
+namespace test {
+using ov::test::utils::PoolingTypes;
+
+std::string PoolingLayerTest::getTestCaseName(const testing::TestParamInfo<poolLayerTestParamsSet>& obj) {
+    poolSpecificParams pool_params;
+    ov::element::Type model_type;
+    std::vector<InputShape> shapes;
+    std::string targetDevice;
+    std::tie(pool_params, model_type, shapes, targetDevice) = obj.param;
+    PoolingTypes pool_type;
+    std::vector<size_t> kernel, stride;
+    std::vector<size_t> pad_begin, pad_end;
+    ov::op::PadType pad_type;
+    ov::op::RoundingType rounding_type;
+    bool excludePad;
+    std::tie(pool_type, kernel, stride, pad_begin, pad_end, rounding_type, pad_type, excludePad) = pool_params;
+
+    std::ostringstream result;
+    result << "IS=(";
+    for (size_t i = 0lu; i < shapes.size(); i++) {
+        result << ov::test::utils::partialShape2str({shapes[i].first}) << (i < shapes.size() - 1lu ? "_" : "");
+    }
+    result << ")_TS=";
+    for (size_t i = 0lu; i < shapes.front().second.size(); i++) {
+        result << "{";
+        for (size_t j = 0lu; j < shapes.size(); j++) {
+            result << ov::test::utils::vec2str(shapes[j].second[i]) << (j < shapes.size() - 1lu ? "_" : "");
+        }
+        result << "}_";
+    }
+    switch (pool_type) {
+        case PoolingTypes::MAX:
+            result << "MaxPool_";
+            break;
+        case PoolingTypes::AVG:
+            result << "AvgPool_";
+            result << "ExcludePad=" << excludePad << "_";
+            break;
+    }
+    result << "K" << ov::test::utils::vec2str(kernel) << "_";
+    result << "S" << ov::test::utils::vec2str(stride) << "_";
+    result << "PB" << ov::test::utils::vec2str(pad_begin) << "_";
+    result << "PE" << ov::test::utils::vec2str(pad_end) << "_";
+    result << "Rounding=" << rounding_type << "_";
+    result << "AutoPad=" << pad_type << "_";
+    result << "netPRC=" << model_type.get_type_name() << "_";
+    result << "trgDev=" << targetDevice;
+    return result.str();
+}
+
+void PoolingLayerTest::SetUp() {
+    poolSpecificParams pool_params;
+    std::vector<InputShape> shapes;
+    ov::element::Type model_type;
+    std::tie(pool_params, model_type, shapes, targetDevice) = this->GetParam();
+    PoolingTypes pool_type;
+    std::vector<size_t> kernel, stride;
+    std::vector<size_t> pad_begin, pad_end;
+    ov::op::PadType pad_type;
+    ov::op::RoundingType rounding_type;
+    bool excludePad;
+    std::tie(pool_type, kernel, stride, pad_begin, pad_end, rounding_type, pad_type, excludePad) = pool_params;
+    init_input_shapes(shapes);
+
+    auto param = std::make_shared<ov::op::v0::Parameter>(model_type, inputDynamicShapes.front());
+
+    std::shared_ptr<ov::Node> pooling = ngraph::builder::makePooling(param,
```

Comment:
`ngraph`?

Reply:
Builders are still in ngraph namespace. Move them after test refactoring